bluemira/codes/_nlopt_api.py

```diff
@@ -384,6 +384,26 @@ def add_eq_constraint(self, f_constraint, tolerance):
         self._opt.add_equality_constraint(f_constraint, tolerance)
         self._append_constraint_tols(f_constraint, tolerance)
 
+    @_opt_inputs_ready
+    def add_eq_constraint(self, f_constraint, tolerance):
+        """
+        Add a vector-valued equality constraint.
+
+        Parameters
+        ----------
+        f_constraint: callable
+            Constraint function
+        tolerance: float
+            Tolerance with which to enforce the constraint
+        """
+        if self.algorithm_name not in EQ_CON_ALGS:
+            raise OptUtilitiesError(
+                f"{self.algorithm_name} does not support equality constraints."
+            )
+
+        self._opt.add_equality_mconstraint(f_constraint, tolerance)
```

Comment:
```suggestion
        self._opt.add_equality_constraint(f_constraint, tolerance)
```

Reply:
nah this is right; its NLOPT for add matrix equality constraint